duckduckgo/Android
# Notify users of async errors

Task/Issue URL: https://app.asana.com/0/1149059203486286/1207045332341271/f

### Description
Should notify users about async errors when syncing.

This PR includes:
 - notifying about bad request / bad format errors

It will also include stacked PRs once they are merged onto this one.

### Steps to test this PR

_Feature 1_
- [x] create sync account
- [x] add few bookmarks and logins
- [x] no warnings inside sync settings, no notifications
- [x] If you don't have notifications permissions enabled, enable them
- [x] apply patch attached to the asana task
- [x] install the version
- [x] make some changes on bookmarks and logins
- [x] ensure warning message is shown in settings and you received a notification
- [x] every time you enter in sync settings, we trigger sync, that should produce a random error which will replace existing one if any.
- [x] Validate error copies

If limit exceeded:
- message should say: Sync paused - You've reached the maximum number of (bookmarks|passwords). Please delete some to resume sync.
- notification should say: (bookmarks|passwords) Sync is Paused -  You've reached the maximum number of (bookmarks|passwords). Please delete some (bookmarks|passwords) to resume sync.

If it's bad request:
- message should say: Sync paused -  Some (bookmarks|passwords) are formatted incorrectly or too long and were not synced. 
- notification should say: (bookmarks|passwords) Sync is Paused -  Some (bookmarks|passwords) are formatted incorrectly or too long and were not synced. 


### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|
